docs(evaluate-artifacts): refactor into 2 pags DOC-347 (#536)
armory/docs: content/en/docs/armory-admin/evaluate-artifacts-stage-enable.md

---
title: Enable the Evaluate Artifacts Stage
description: >
  Enable the Evaluate Artifacts Stage plugin to add it to Armory Enterprise. The stage gives pipeline creators the ability to create base64 artifacts from text saved within a pipeline. The contents of the new artifact include the results of any SpEL expressions that are evaluated.
aliases:
  - /docs/plugin-guide/plugin-evaluate-artifacts/
---

![Proprietary](/images/proprietary.svg)

## Overview

Some artifact types and their associated stages, such as Kubernetes, support  using SpEL to read parameters from the Spinnaker context and inject them into an artifacts manifest at deploy time. Other stages, such as the Terraform Integration stage, do not support this out of the box. The Evaluate Artifacts Stage plugin adds supports for evaluating SpEL against any artifact within your Spinnaker pipeline.

> For information about how to use the stage, see [Use the Evaluate Artifacts Stage]({{< ref "evaluate-artifacts-stage-use.md" >}}).

Reviewer: ```suggestion
For information about how to use the stage, see {{< linkWithTitle "evaluate-artifacts-stage-use.md" >}}.
```

You can use the `linkWithTitle` shortcode if you don't want to duplicate the page title in a normal link

         evaluateArtifacts:
           url: https://raw.githubusercontent.com/armory-plugins/evaluate-artifacts-releases/master/repositories.json
   ```

   Make sure to replace `PLUGIN_VERSION` with the version of the plugin you want to use. Plugin versions can be found [here](#versions).

2. Enable the UI for the stage in your `~/.hal/default/profiles/gate-local.yml` file:

   ```yaml
   spinnaker:
     extensibility:
      deck-proxy:
        enabled: true
        plugins:
          Armory.EvaluateArtifactsPlugin:
            enabled: true
            version: <PLUGIN_VERSION> # Replace with the version you want to use. For example, use 0.1.0.
      repositories:
        evaluateArtifacts:
          url: https://raw.githubusercontent.com/armory-plugins/evaluate-artifacts-releases/master/repositories.json
   ```

   Make sure to replace `PLUGIN_VERSION` with the version of the plugin you want to use. Plugin versions can be found [here](#versions).

3. Run `hal deploy apply` to apply the changes to your Armory Enterprise instance.

{{% /tab %}}
{{< /tabs>}}

## Versions

- 0.1.0 - Improved the user experience. Execution errors for the stage now display in the UI.

> - 0.0.10 - Initial Release

Reviewer: ```suggestion
- 0.0.10 - Initial Release

## {{% heading "nextSteps" %}}

* {{< linkWithTitle "evaluate-artifacts-stage-use.md" >}}
```

Reviewer: This type of linking helps with SEO also

---
title: Enable the Evaluate Artifacts Stage Plugin
description: >
  Enable the Evaluate Artifacts Stage plugin to add a stage to Armory Enterprise that makes it easy to evaluate SpEL expressions inside of Spinnaker™ artifacts.
aliases:
  - /docs/plugin-guide/plugin-evaluate-artifacts/
---

![Proprietary](/images/proprietary.svg)

## Overview

> Armory’s Evaluate Artifacts plugin allows you to easily evaluate SpEL queries inside of any Spinnaker artifacts. Some artifact types and their associated stages, such as Kubernetes, support  using SpEL to read parameters from the Spinnaker context and inject them into an artifacts manifest at deploy time. Other stages, such as the Terraform Integration stage, do not support this out of the box. The Evaluate Artifacts Stage plugin adds supports for evaluating SpEL against any artifact within your Spinnaker pipeline.

Reviewer 2: There's a double space after `Kubernetes, support  `

For information about how to use the stage, see [Use the Evaluate Artifacts Stage]({{< ref "evaluate-artifacts-plugin-use.md" >}}).

> ## Setup

Reviewer 2: Question: should we add Halyard setup too?

Reviewer 3: yeah..... we should.

---
title: Enable the Evaluate Artifacts Stage Plugin
description: >
  Enable the Evaluate Artifacts Stage plugin to add a stage to Armory Enterprise that makes it easy to evaluate SpEL expressions inside of Spinnaker™ artifacts.
aliases:
  - /docs/plugin-guide/plugin-evaluate-artifacts/
---

> ![Proprietary](/images/proprietary.svg)

Reviewer: Does this mean the plugin only works with Armory Enterprise?

Author: It means it's a proprietary plugin and is not open source

Reviewer: Ok. Stephen's blog post says it works with AE and Spinnaker, so seeing the Proprietary graphic made me think it only works with AE.

## Overview

Armory’s Evaluate Artifacts plugin allows you to easily evaluate SpEL queries inside of any Spinnaker artifacts. Some artifact types and their associated stages, such as Kubernetes, support  using SpEL to read parameters from the Spinnaker context and inject them into an artifacts manifest at deploy time. Other stages, such as the Terraform Integration stage, do not support this out of the box. The Evaluate Artifacts Stage plugin adds supports for evaluating SpEL against any artifact within your Spinnaker pipeline.

For information about how to use the stage, see [Use the Evaluate Artifacts Stage]({{< ref "evaluate-artifacts-plugin-use.md" >}}).

> 

Reviewer: Before `Setup`, what do you think of including a _Before you begin_  section that points to the OSS plugin use guide https://spinnaker.io/guides/user/plugins/ ? The first few sections cover terminology, but the latter half details how to install a plugin using Halyard.

Author: I have the directions for Halyard written up but haven't pushed them yet. But I'll add the link in the pre-reqs so people can reference the terminology.

## Setup

> Add the following snippet to your Operator config, such as `spinnakerservice.yml`:

Reviewer: @brian-armory `spinnakerservice.yml` isn't the Operator config. It's the Spinnaker manifest.  I clarified the terminology in the Operator docs PR.

{{< prism lang=yaml line="4-5, 10, 22" >}}
    spec:
      spinnakerConfig:
        profiles:
          spinnaker:  
            spinnaker:
              extensibility:
                plugins:
                  Armory.EvaluateArtifactsPlugin:
                    enabled: true
                    version: <PLUGIN_VERSION> # Replace with the version you want to use
              repositories:
                evaluateArtifacts:
                  url: https://raw.githubusercontent.com/armory-plugins/evaluate-artifacts-releases/master/repositories.json
          gate:
            spinnaker:
              extensibility:
                deck-proxy:
                  enabled: true
                  plugins:
                    Armory.EvaluateArtifactsPlugin:
                      enabled: true
                      version: <PLUGIN_VERSION> # Replace with the version you want to use. Omit the 'v' when providing a version. For example, use 0.1.0, not v0.1.0
{{< /prism >}}

Keep the following in mind when using this configuration snippet: 

> * Make sure to include the nested `spinnaker` parameters on lines 4 and 5. Both are required because of how Operator configs are processed and consumed by Armory Enterprise.

Reviewer: ```suggestion
* Make sure to include the nested `spinnaker` parameters on lines 4 and 5. Both are required because of how Spinnaker consumes plugin configuration.
```

* Replace `<PLUGIN_VERSION>` on lines 10 and 22 with the version of the plugin that you want to use. Plugin versions can be found [here](#versions).

Then, deploy your updated Operator configuration using one of the following methods:

- The `deploy.sh` script if you use the [Armory kustomize repo](https://github.com/armory/spinnaker-kustomize-patches) to help manage your Operator configurations

> - `kubectl -n <spinnaker-namespace> -f <operator-config>.yml​`

Reviewer: You are deploying an Armory Enterprise manifest, not an updated Operator configuration. 

My understanding of `deploy.sh` is that it's for a quickstart deployment only, that you wouldn't use it for deploying an updated Armory Enterprise manifest. How you deploy Spinnaker or AE depends on whether you are using a single `spinnakerservice.yml` manifest file or are using Kustomize patches.  

If using a single manifest file:
```bash
kubectl -n <namespace> apply -f <path-to-manifest-file>
```

If using Kustomize patches:
```bash
kubctl -n <namespace> apply -k <path-to-kustomize-directory>
```
or if you have Kustomize installed:

```bash
# the namespace is declared in kustomization.yml
# run this from the same directory as your kustomization.yml file
kustomize build | kubectl  apply -f -
```
